src/tests/Application.vala

```diff
@@ -145,7 +145,7 @@ namespace Terminal.Test.Application {
                 unowned var window = (MainWindow) application.active_window;
                 assert_nonnull (window);
                 var n_tabs = (int) window.terminals.length ();
-                assert_cmpint (n_tabs, CompareOperator.EQ, 2);
+                assert_cmpint (n_tabs, CompareOperator.EQ, 1); // No default tab added as well in this case
```

Comment:
this test was made to assert that we always create a new tab when requested, so this change is wrong.

Reply:
It is necessary for the test to pass.  If the original test is correct then the app is wrong

Reply:
If you run `io.elementary.terminal -t` manually without a Terminal window open you get one tab (as expected). If you run the same command with the primary window open (but no restored tabs) then you get 2 tabs - the original default one and the requested additional one.

I am not exactly sure what happens in the test environment - does a window get created before the options are processed or not?

Reply:
After some investigation it seems like before processing the commandline in `option ()` there is 1 window with 1 tab. So I would have expected `-t` true to create a new tab like when used manually.  But it doesn't in the test environment for some reason.

Reply:
OK, so it is race - the callback is executed before the window terminal list is updated in the `on_tab_added` signal handler.  Need to get a more immediate count of true number of tabs in the notebook.  I'll review whether the terminals list is needed or not.